net/ip.go

```diff
@@ -79,11 +79,13 @@ func IPFromString(str string) (IP, error) {
 
 // Equal returns true if ip is equal to other
 func (ip IP) Equal(other IP) bool {
-	return ip == other
```

Comment:
Isn't the existing code the shorter version of the one you're proposing here?

Reply:
doesn't this imply much more "magic"?
I thought having the comparison more verbose is much easier to understand what's compared instead of relying on some interpreter magic. What's compared here? Are those pointer values which are compared, is it a bit-wise diff of the objects, are the `Equal` functions of each object are called implicitly?

Reply:
ack. Changed it back as it was